Source/ARTDefault.m

```diff
@@ -37,8 +39,24 @@ @implementation ARTDefault
 static NSTimeInterval _connectionStateTtl = 60.0;
 static NSInteger _maxMessageSize = 65536;
 
++ (NSArray*)fallbackHostsWithEnvironment:(NSString *)environment {
+    NSString *prefix = @"";
+    NSString *suffix = @"";
+    if (environment && ![[environment stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]] isEqualToString:@""] && ![environment isEqualToString:ARTDefaultProduction]) {
+        prefix = [NSString stringWithFormat:@"%@-", environment];
+        suffix = @"-fallback";
+    }
+    return @[
+        [NSString stringWithFormat:@"%@a%@.ably-realtime.com", prefix, suffix],
+        [NSString stringWithFormat:@"%@b%@.ably-realtime.com", prefix, suffix],
+        [NSString stringWithFormat:@"%@c%@.ably-realtime.com", prefix, suffix],
+        [NSString stringWithFormat:@"%@d%@.ably-realtime.com", prefix, suffix],
+        [NSString stringWithFormat:@"%@e%@.ably-realtime.com", prefix, suffix]
+    ];
```

Comment:
Since it could be iterable, this array should be wrapped in a loop.

Reply:
I think it's better to use objc `map` equivalent: `valueForKey:` - https://stackoverflow.com/a/6127874/961037

Reply:
changed here 7e9b470fe1d5bc51d38743621975a67ba6f8f220